code-challenges.js

```diff
@@ -69,7 +116,45 @@ var vowelTester2 = "throw"
 var websites = ["codepen.io", "codecademy.com", "coursera.org", "codepen.io", "udemy.com", "pluralsight.com", "udacity.com", "sitepoint.com"]
 // Expected output: "codepen.io" "codepen.io"
 
+const findIoSites = (array) => {
 
+  // array to hold our found io sites
+  let ioSitesArray = []
+
+  // map over the array and access each website
+  array.map(currentSite => {
+
+    let vowelIndex = 0
+    // first find the index of the '.'
+    for(let i = 0; i < currentSite.length; i++) {
+      // if you find the '.' and the next indexed value is 'i'
+      if (currentSite[i] === '.' && currentSite[i + 1] === 'i') {
+        // assume we found an .io website
+        ioSitesArray.push(currentSite)
+        break
+      } // end if
+    } // end for
+  }); // end .map
+
+  return ioSitesArray
+}
+
+console.log(findIoSites(websites));
 
 
 // --------------------6) STRETCH: Write a function that takes a number as an argument and uses a WHILE loop to count up to that number from 0.
+
+const countUsingWhile = (number) => {
+  // need a counter for the loops
+  let count = 0
+  // need an array for the values
+  let numberArray = [0]
+  // loop until we reach our number
+  while(count++ < number) {
+    // keep track of the numbers
+    numberArray.push(count);
+  }
+  return numberArray;
+}
+
```

Comment:
This totally works great! I'd maybe refacort this to see how we can do it without having to push into an array.